### PR TITLE
Fix alias typing and tests

### PR DESCRIPTION
**Pull request by nolag**

### CODE REVIEW

File: pkg/values/value_test.go [hunk 1]
```diff
@@ -228,6 +229,41 @@ func Test_Value(t *testing.T) {
...
+		{name: "uint64", test: func(tt *testing.T) { wrappableTest[int64, uint64](tt, anyValue) }},
+		{name: "uint32", test: func(tt *testing.T) { wrappableTest[int64, uint32](tt, anyValue) }},
+		{name: "uint16", test: func(tt *testing.T) { wrappableTest[int64, uint16](tt, anyValue) }},
+		{name: "uint8", test: func(tt *testing.T) { wrappableTest[int64, uint8](tt, anyValue) }},
+		{name: "uint", test: func(tt *testing.T) { wrappableTest[int64, uint](tt, anyValue) }},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(st *testing.T) {
+
```
Comment (kidambisrinivas) [inline pkg/values/value_test.go:262]: Should we call tc.test?

File: pkg/values/big_int.go [hunk 2]
```diff
@@ -31,9 +30,10 @@ func (b *BigInt) Unwrap() (any, error) {
 
 func (b *BigInt) UnwrapTo(to any) error {
 	if b == nil || b.Underlying == nil {
-		return errors.New("could not unwrap nil values.BigInt")
+		return fmt.Errorf("could not unwrap nil")
```
Comment (jmank88) [inline pkg/values/big_int.go:33]: The linter should be telling you to use `errors.New` actually.
```suggestion
		return errors.New("could not unwrap nil")
```

**Review by kidambisrinivas — APPROVED** [review]

**Review by kidambisrinivas — APPROVED** [review]

**Review by cedric-cordenier — APPROVED** [review]

**Comment by cedric-cordenier** [comment]
LGTM, but we need to be careful to resolve the merge conflicts with my changes to int unwrapping which moves to using unwrapTo + reflection for bounds checking @kidambisrinivas

**Comment by nolag** [comment]
> LGTM, but we need to be careful to resolve the merge conflicts with my changes to int unwrapping which moves to using unwrapTo + reflection for bounds checking @kidambisrinivas

The way I have it in this PR is a lot cleaner for dealing with aliases.  I don't want unwrapTo to need to know about int types.

**Comment by cedric-cordenier** [comment]
@nolag It doesn't: all unwrapTo knows about is how to deal with aliases, the bounds checking is handled outside of it

File: pkg/values/int.go [hunk 3]
```diff
@@ -41,27 +42,116 @@ func (i *Int64) UnwrapTo(to any) error {
 		return fmt.Errorf("cannot unwrap to nil pointer: %+v", to)
 	}
 
-	if reflect.ValueOf(to).Kind() != reflect.Pointer {
-		return fmt.Errorf("cannot unwrap to non-pointer value: %+v", to)
-	}
+	switch tv := to.(type) {
```
[the remark this answers is not in the document]
Comment (cedric-cordenier) [reply pkg/values/int.go:45]: Discussed offline: we're going to go with this for now given the time crunch

**Review by ettec — APPROVED** [review]

File: pkg/values/value_test.go [hunk 4]
```diff
@@ -383,81 +419,110 @@ func Test_Copy(t *testing.T) {
...
-			name:  "simple aliases",
-			val:   func() any { return "string" },
-			alias: func() any { return aliasString("") },
+			name: "byte alias in slice",
+			test: func(tt *testing.T) {
+				wrappableTestWithConversion[[]byte, []aliasByte](tt, []byte("hello"), func(native []aliasByte) []byte {
+					converted := make([]byte, len(native))
+					for i, b := range native {
+						converted[i] = byte(b)
+					}
```
[the remark this answers is not in the document]
Comment (jmank88) [reply pkg/values/value_test.go:447]: Actually I think the compiler knows to do this

**Pull request merged**